api/lib/opentelemetry/trace/samplers/never_sample_sampler.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/trace/samplers/basic_decision'
+
+module OpenTelemetry
+  module Trace
+    module Samplers
+      # The {NeverSampleSampler} always returns a false sampling decision.
+      class NeverSampleSampler
+        NEVER_SAMPLE_DECISION = BasicDecision.new(decision: false)
+
+        # Returns the sampling {Decision} for a {Span} to be created.
+        #
+        # @param [SpanContext] span_context The {SpanContext} of a parent span,
+        #   typically extracted from the wire. Can be nil.
+        # @param [Boolean] extracted_context True if span_context was extracted
+        #   from the wire. Can be nil.
+        # @param [Integer] trace_id The trace_id of the {Span} to be created
+        # @param [Integer] span_id The span_id of the {Span} to be created
+        # @param [String] span_name Name of the {Span} to be created
+        # @param [Array<Links>] links A collection of links to be associated
+        #   with the {Span} to be created
+        # @return [Decision] The sampling decision
+        def should_sample(span_context: nil,
```

Comment:
This feels "almost" like a predicate, which leads to questions like: is `should_` a common prefix for a predicate in Ruby? Should this end with `?`? It's a little annoying, and I wish we could rename the method to be more descriptive of the return value. It *would* be a boolean if we didn't also want to provide attributes describing the decision.

Reply:
Originally, I was treating this as a predicate method and had the `?`, however, since it doesn't return a boolean value, it's really not a predicate, so I decided remove it. Since it returns a `Decision` instance, it really can't be used like a predicate either. You really need to assign to the return value, and then check the return value. Or chain through to it: `sampler.should_sample(...).sampled?`. I'm definitely not loving the API overall.

Reply:
I would be open to other names that stick to the spirit of the spec, but are more idiomatic. Possibly calling it `sampling_decision(...)` or `sampling_decision_for(...)`would be slight improvements. I'm open to other ideas though.

Reply:
Either of `decision(...)` or `decision_for(...)` would be 👍 for me. The tests don't read well at the moment:
```ruby
sampler.should_sample(...).sampled?
```
vs.
```ruby
sampler.sampling_decision(...).sampled?
```
vs.
```ruby
sampler.sampling_decision_for(...).sampled?
```
vs.
```ruby
sampler.decision(...).sampled?
```
vs.
```ruby
sampler.decision_for(...).sampled?
```

In most cases, the sampler will be called something like `sampler`, so the shorter method name reduces stuttering a little.